paddle/fluid/operators/optimizers/merged_momentum_op.h

```diff
@@ -182,14 +240,82 @@ class MergedMomentumOpKernel : public framework::OpKernel<T> {
     VLOG(10) << "Launch MergedMomentum kernel " << i << " "                  \
              << kernel_params.param_num;                                     \
   }
-
-    if (multi_precision) {
-      PADDLE_LAUNCH_MERGED_MOMENTUM_KERNEL(true);
+      if (multi_precision) {
+        PADDLE_LAUNCH_MERGED_MOMENTUM_KERNEL(true);
+      } else {
+        PADDLE_LAUNCH_MERGED_MOMENTUM_KERNEL(false);
+      }
+#undef PADDLE_LAUNCH_MERGED_MOMENTUM_KERNEL
     } else {
-      PADDLE_LAUNCH_MERGED_MOMENTUM_KERNEL(false);
-    }
+      for (size_t idx = 0; idx < n; idx++) {
+        std::string regularization_method = "";
+        if (regularization_methods.size() != 0) {
+          regularization_method = regularization_methods[idx];
+        }
+        RegularizationType regularization_flag{RegularizationType::kNONE};
```

Comment:
NIT. Not required to change. Maybe the following code would be simpler:
```c++
RegularizationType regularization_flag = regularization_methods.size() > 0 &&  regularization_methods[idx] == "l2_decay" ? RegularizationType::kL2DECAY : RegularizationType::kNONE.
```

Reply:
tks, this code has been modified according to the comments.